civis/ml/_model.py

```diff
@@ -268,6 +257,132 @@ def _show_civisml_warnings(warn_list):
             warnings.warn(warn_str, RuntimeWarning)
 
 
+def _get_job_type_version(alias):
+    """Derive the job type and version from the given alias.
+
+    Parameters
+    ----------
+    alias : str
+        CivisML alias
+
+    Returns
+    -------
+    str
+        Job type, one of {training, prediction, registration}.
+    str
+        CivisML version, e.g., "v2.2".
+    """
+    # A version-less alias for production, e.g., "civis-civisml-training"
+    match_production = re.search(r'\Acivis-civisml-(\w+)\Z', alias)
+    # A versioned alias, e.g., "civis-civisml-training-v2-3"
+    match_v = re.search(r'\Acivis-civisml-(\w+)-v(\d+)-(\d+)\Z', alias)
+    # A special-version alias, e.g., "civis-civisml-training-dev"
+    match_special = re.search(r'\Acivis-civisml-(\w+)-(\S+[^-])\Z', alias)
+
+    if match_production:
+        job_type = match_production.group(1)
+        version = None
+    elif match_v:
+        job_type = match_v.group(1)
+        version = 'v%s.%s' % match_v.group(2, 3)
+    elif match_special:
+        job_type = match_special.group(1)
+        version = match_special.group(2)
+    else:
+        msg = ('Unable to parse the job type and version '
+               'from the CivisML alias "%r"')
+        raise ValueError(msg % alias)
```

Comment:
Previously this ValueError message said "it doesn't look like a CivisML alias". This message was misleading, because the alias wouldn't end up being here if it didn't start with `"civis-civisml-"` from the upstream code.